include/libopencm3/usb/usbstd.h

```diff
@@ -163,7 +163,8 @@ struct usb_config_descriptor {
 
 	/* Descriptor ends here.  The following are used internally: */
 	const struct usb_interface {
-		int num_altsetting;
+		uint8_t * cur_altsetting;
+		uint8_t num_altsetting;
```

Comment:
are these uint8_t to save ram? or are they actually limited in some way?  just thinking out loud, but ```int_least8_t``` might be enough here?